sentry-android-replay/src/main/java/io/sentry/android/replay/capture/SessionCaptureStrategy.kt

```diff
@@ -98,20 +95,24 @@ internal class SessionCaptureStrategy(
 
             val now = dateProvider.currentTimeMillis
             if ((now - currentSegmentTimestamp.time >= options.sessionReplay.sessionSegmentDuration)) {
-                val segment =
-                    createSegmentInternal(
-                        options.sessionReplay.sessionSegmentDuration,
-                        currentSegmentTimestamp,
-                        currentReplayId,
-                        currentSegment,
-                        height,
-                        width
-                    )
-                if (segment is ReplaySegment.Created) {
-                    segment.capture(scopes)
-                    currentSegment++
-                    // set next segment timestamp as close to the previous one as possible to avoid gaps
-                    segmentTimestamp = segment.replay.timestamp
+                recorderConfig?.let { config ->
```

Comment:
can we capture values of the recorderConfig on the main thread (basically before replayExecutor.submit) and then pass them to the runnable? 

I'm thinking of a case where recorderConfig changes (e.g. orientation change), and we'd be trying to capture a segment with the new values already, however the screenshots would be still from before orientation change